src/promptflow-evals/promptflow/evals/evaluate/_utils.py

```diff
@@ -199,3 +202,13 @@ def _apply_column_mapping(source_df: pd.DataFrame, mapping_config: dict, inplace
 
 def _has_aggregator(evaluator):
     return hasattr(evaluator, "__aggregate__")
+
+
+def set_event_loop_policy():
+    import asyncio
+    import platform
+
+    if platform.system().lower() == "windows":
+        # Reference: https://stackoverflow.com/questions/45600579/asyncio-event-loop-is-closed-when-getting-loop
+        # On Windows seems to be a problem with EventLoopPolicy, use this snippet to work around it
+        asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())

```

Comment:
According to the reference link, seems like this is legacy issue and after python 3.7, we could use `asyncio.run` instead of `loop.run_until_complete()` to fix it.

The only `loop.run_until_complete()` usage is in adversarial_simulator.py, maybe we should update that file and skip is platform specific change.

https://github.com/microsoft/promptflow/blob/4f7f41f7e36f8a31867f9354d9a00cb3005540d4/src/promptflow-evals/promptflow/evals/synthetic/adversarial_simulator.py#L372

![image](https://github.com/user-attachments/assets/540eae3f-b29c-4bfa-8881-2be1b78037f5)